sdk/Pulumi.Tests/Resources/ComponentResourceTests.cs

```diff
@@ -112,5 +171,16 @@ public Task<object> CallAsync(MockCallArgs args)
         {
             return (args.Name + "-id", args.Inputs);
         }
+
+        public Task RegisterResourceOutputs(MockRegisterResourceOutputsRequest request)
+        {
+            _registeredOutputsByUrn[request.Urn ?? ""] = request.Outputs;
+            return Task.CompletedTask;
+        }
+
+        public ImmutableDictionary<string, object> GetRegisteredOutputs(string urn)
```

Comment:
Should we just give a read-only view access to the whole dictionary? Then users can either decided to lookup by URN or just iterate over all the resources.

Reply:
This is specific to the unit test in question and how it implemented `IMocks`, nothing user-facing

Reply:
Ah yes, my oversight, that's all good then.